elements/lisk-api-client/src/ipc_channel.ts

```diff
@@ -34,7 +35,8 @@ import { convertRPCError } from './utils';
 const CONNECTION_TIME_OUT = 2000;
 
 const getSocketsPath = (dataPath: string) => {
-	const socketDir = path.join(path.resolve(dataPath), 'tmp', 'sockets');
+	const dataPathWithoutTilde = dataPath.replace('~', homedir());
+	const socketDir = path.join(path.resolve(dataPathWithoutTilde), 'tmp', 'sockets');
```

Comment:
```suggestion
	const socketDir = path.join(dataPath.replace('~', homedir()), 'tmp', 'sockets');
```

Reply:
I think this will not work for relative data paths like `./myDir`.
Although can be reformatted as 
`const socketDir = path.join(path.resolve(dataPath.replace('~', homedir())), 'tmp', 'sockets');`